### PR TITLE
Add addtionalArgs for proxy-server and proxy-agent

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

### CODE REVIEW

File: pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go [hunk 1]
```diff
@@ -287,6 +291,9 @@ type ManagedProxyConfigurationProxyAgent struct {
 	// +optional
 	// +kubebuilder:default=3
 	Replicas int32 `json:"replicas"`
+	// `additionalArgs` defines args used in proxy-agent.
+	// +optional
+	AdditionalArgs []string `json:"additionalArgs"`
```
Comment (yue9944882) [inline pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go:296]: nit: omitempty

File: pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go [hunk 2]
```diff
@@ -204,6 +204,10 @@ type ManagedProxyConfigurationProxyServer struct {
 	// `entrypoint` defines how will the proxy agents connecting the servers.
 	// +optional
 	Entrypoint *ManagedProxyConfigurationProxyServerEntrypoint `json:"entrypoint"`
+
+	// `additionalArgs` defines args used in proxy-server.
```
Comment (yue9944882) [inline pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go:208]: // `additionalArgs` adds arbitrary additional command line args to the proxy-server.

File: pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go [hunk 3]
```diff
@@ -204,6 +204,10 @@ type ManagedProxyConfigurationProxyServer struct {
...
+	// +optional
+	AdditionalArgs []string `json:"additionalArgs"`
```
Comment (yue9944882) [inline pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go:210]: nit: omitempty

**Review by yue9944882 — COMMENTED** [review]
@xuezhaojun a few nits otherwise lgtm

File: pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go (hunk 1 above)
Comment (xuezhaojun) [reply pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go]: Added.

File: pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go (hunk 2 above)
Comment (xuezhaojun) [reply pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go]: Changed the description.

File: pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go (hunk 3 above)
Comment (xuezhaojun) [reply pkg/apis/proxy/v1alpha1/managedproxyconfiguration_types.go]: Added.

**Review by yue9944882 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**